bin/martinize2

```diff
@@ -140,42 +141,27 @@ def martinize(system, mappings, to_ff, delete_unknown=False):
     return system
 
 
-def write_gmx_topology(system, top_path, deduplicate=True, header=()):
+def write_gmx_topology(system, top_path, header=()):
     """
     Writes a Gromacs .top file for the specified system.
     """
     if not system.molecules:
         raise ValueError('No molecule in the system. Nothing to write.')
-    if deduplicate:
-        # Deduplicate the moleculetypes in order to write each molecule ITP only
-        # once.
-        molecule_types = [[system.molecules[0], [system.molecules[0]]], ]
-        for molecule in system.molecules[1:]:
-            for molecule_type, share_moltype in molecule_types:
-                if molecule.share_moltype_with(molecule_type):
-                    share_moltype.append(molecule)
-                    break
-            else:  # no break
-                molecule_types.append([molecule, [molecule, ]])
-    else:
-        molecule_types = [[molecule, [molecule, ]] for molecule in system.molecules]
+
     # Write the ITP files for the moleculetypes.
-    for molidx, (molecule_type, _) in enumerate(molecule_types):
-        molecule_type.moltype = 'molecule_{}'.format(molidx)
-        with open('molecule_{}.itp'.format(molidx), 'w') as outfile:
-            vermouth.gmx.itp.write_molecule_itp(molecule_type, outfile, header=header)
-    # Reorganize the molecule type assignment to write the top file.
-    # The top file "molecules" section lists the molecules in the same order
-    # as in the structure and group them. To do the grouping, we associate each
-    # molecule to the molecule type (its name actually) instead of associating
-    # the molecule types with the molecules as we did above.
-    molecule_to_type = {}
-    for molecule_type, share_moltype in molecule_types:
-        for molecule in share_moltype:
-            molecule_to_type[molecule] = molecule_type.moltype
+    max_name_length = 0
+    moltype_count = OrderedDict()
+    molecule_groups = itertools.groupby(system.molecules, key=lambda x: x.meta['moltype'])
```

Comment:
Should come with a comment that it should *not* be sorted with the same key, because of reasons